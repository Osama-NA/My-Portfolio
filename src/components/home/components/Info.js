import React from 'react';
import { Link } from 'react-router-dom';
import { InfoContainer } from '../styles/Info.styled';
import { GlobeContainer } from './GlobeContainer';

export const Info = ({circleBackground}) => {
    return (
        <InfoContainer data-testid="info">
            <h1>Osama Al Hajj Ali</h1>
            <h3>I am a web developer</h3>

            <nav>
                <ul data-testid="nav-bar">
                    <li><Link to="/Router/Projects" data-testid="projects-link">Projects</Link></li>
                    <li><Link to="/Router/Contacts" data-testid="contacts-link">Contact me</Link></li>
                    <li><Link to="/Router/About" data-testid="about-link">About me</Link></li>
                </ul>
            </nav>

            <GlobeContainer circleBackground={circleBackground} />
        </InfoContainer>
    )
}