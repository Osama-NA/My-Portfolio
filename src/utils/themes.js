// GREEN / GRAY
export const greenGrayTheme = {
    themeColors: {
        mainColor: '#54EBA0'
    },
    fontColors: {
        secondary: '#e0e0e0',
        primary: '#fff',
        gray: '#5e5e5e'
    },
    backgroundColors: {
        lightSide: '#313138',
        darkSide: '#242429',
        lightBackground: '#3F3F47',
        navBackground: '#383840'
    },
    cubesColor: '#03AD78',
    circleBackground: '#313138'
}

// BLUEISH PURPLE / GRAY 
export const blueishPurpleGrayTheme = {
    themeColors: {
        mainColor: '#674DFF'
    },
    fontColors: {
        secondary: '#e0e0e0',
        primary: '#fff',
        gray: '#5e5e5e'
    },
    backgroundColors: {
        lightSide: '#313138',
        darkSide: '#242429',
        lightBackground: '#3F3F47',
        navBackground: '#383840'
    },
    cubesColor: '#544DFF',
    circleBackground: '#313138'
}

// PURPLE / GRAY
export const purpleGrayTheme = {
    themeColors: {
        mainColor: '#B15EFF'
    },
    fontColors: {
        secondary: '#e0e0e0',
        primary: '#fff',
        gray: '#5e5e5e'
    },
    backgroundColors: {
        lightSide: '#313138',
        darkSide: '#242429',
        lightBackground: '#3F3F47',
        navBackground: '#383840'
    },
    cubesColor: '#9A52DE',
    circleBackground: '#313138'
}

// LIGHT BLUE / GRAY
export const lightBlueGrayTheme = {
    themeColors: {
        mainColor: '#00E6E2'
    },
    fontColors: {
        secondary: '#e0e0e0',
        primary: '#fff',
        gray: '#5e5e5e'
    },
    backgroundColors: {
        lightSide: '#313138',
        darkSide: '#242429',
        lightBackground: '#3F3F47',
        navBackground: '#383840'
    },
    cubesColor: '#00E6E2',
    circleBackground: '#313138'
}

// YELLOW / GRAY
export const yellowGrayTheme = {
    themeColors: {
        mainColor: '#F2F270'
    },
    fontColors: {
        secondary: '#e0e0e0',
        primary: '#fff',
        gray: '#5e5e5e'
    },
    backgroundColors: {
        lightSide: '#313138',
        darkSide: '#242429',
        lightBackground: '#3F3F47',
        navBackground: '#383840'
    },
    cubesColor: '#F2F208',
    circleBackground: '#313138'
}

// PINK / GRAY
export const pinkGrayTheme = {
    themeColors: {
        mainColor: '#FF558E'
    },
    fontColors: {
        secondary: '#e0e0e0',
        primary: '#fff',
        gray: '#5e5e5e'
    },
    backgroundColors: {
        lightSide: '#313138',
        darkSide: '#242429',
        lightBackground: '#3F3F47',
        navBackground: '#383840'
    },
    cubesColor: '#FF558E',
    circleBackground: '#313138'
}

// RED / GRAY
export const redGrayTheme = {
    themeColors: {
        mainColor: '#FF3142'
    },
    fontColors: {
        secondary: '#e0e0e0',
        primary: '#fff',
        gray: '#5e5e5e'
    },
    backgroundColors: {
        lightSide: '#313138',
        darkSide: '#242429',
        lightBackground: '#3F3F47',
        navBackground: '#383840'
    },
    cubesColor: '#FF3142',
    circleBackground: '#313138'
}