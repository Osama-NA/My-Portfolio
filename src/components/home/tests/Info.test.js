import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { BrowserRouter } from 'react-router-dom';
import { Info } from '../components/Info.js';
import { cleanup, render } from '@testing-library/react';
import { lightPinkTheme } from '../../../utils/themes.js';

afterEach(cleanup);

describe("Info component tests", () => {

    const InfoComponent = () => {
        return (
            <BrowserRouter>
                <ThemeProvider theme={lightPinkTheme}>
                    <Info circleBackground="lightblue" />
                </ThemeProvider>
            </BrowserRouter>
        )
    }

    it("Info component renders without crashing", () => {
        let div = document.createElement('div');
        ReactDOM.render(<InfoComponent />, div);
    })

    it("Info component is in the document", () => {
        const { getByTestId } = render(<InfoComponent />);
        expect(getByTestId('info')).toBeInTheDocument();
    })

    it("Info component contains navigation bar", () => {
        const { getByTestId } = render(<InfoComponent />);
        expect(getByTestId('info')).toContainElement(getByTestId('nav-bar'));
    })

    it("Info component contains globe", () => {
        const { getByTestId } = render(<InfoComponent />);
        expect(getByTestId('info')).toContainElement(getByTestId('globe'));
    })

    it("Globe element renders without crashing", () => {
        const { getByTestId } = render(<InfoComponent />);
        expect(getByTestId('info')).toContainElement(getByTestId('globe'));
    })

    it("Navigation bar has projects link", () => {
        const { getByTestId } = render(<InfoComponent />);
        expect(getByTestId('projects-link')).toHaveAttribute('href', '/Router/Projects');
    })

    it("Navigation bar has contacts link", () => {
        const { getByTestId } = render(<InfoComponent />);
        expect(getByTestId('contacts-link')).toHaveAttribute('href', '/Router/Contacts');
    })

    it("Navigation bar has about link", () => {
        const { getByTestId } = render(<InfoComponent />);
        expect(getByTestId('about-link')).toHaveAttribute('href', '/Router/About');
    })

    it("Info component matches snapshot", () => {
        const tree = renderer.create(<InfoComponent />).toJSON();
        expect(tree).toMatchSnapshot();
    })
})