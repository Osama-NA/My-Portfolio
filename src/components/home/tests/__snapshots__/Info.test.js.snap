// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info component tests Info component matches snapshot 1`] = `
<main
  className="sc-bdfBQB kdrJWn"
  data-testid="info"
>
  <h1>
    Osama Al Hajj Ali
  </h1>
  <h3>
    I am a web developer
  </h3>
  <nav>
    <ul
      data-testid="nav-bar"
    >
      <li>
        <a
          data-testid="projects-link"
          href="/Router/Projects"
          onClick={[Function]}
        >
          Projects
        </a>
      </li>
      <li>
        <a
          data-testid="about-link"
          href="/Router/About"
          onClick={[Function]}
        >
          About me
        </a>
      </li>
      <li>
        <a
          data-testid="contacts-link"
          href="/Router/Contacts"
          onClick={[Function]}
        >
          Contact me
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="globe-container"
  >
    <div
      className="globe"
      data-testid="globe"
    >
      <div
        style={
          Object {
            "height": "100%",
            "overflow": "hidden",
            "position": "relative",
            "width": "100%",
          }
        }
      >
        <canvas
          style={
            Object {
              "display": "block",
            }
          }
        />
      </div>
    </div>
  </div>
</main>
`;
